src/tarantool.client/LogicalConnectionWrapper.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Threading.Tasks;
+using ProGaudi.Tarantool.Client.Model.Requests;
+using ProGaudi.Tarantool.Client.Model.Responses;
+
+namespace ProGaudi.Tarantool.Client
+{
+    using System.Threading;
+
+    using ProGaudi.Tarantool.Client.Model;
+    using ProGaudi.Tarantool.Client.Utils;
+
+    public class LogicalConnectionWrapper : ILogicalConnection
```

Comment:
prefix all calls in Box to check connection and reconnect, or implement something like ILogicalConnection (with ability to drop logical connection and reconnect) inside box and make calls to it